tests/integration/integration-test.sh

```diff
@@ -21,21 +21,24 @@ COSIGN_PUBLIC_KEY="$(printf -- "${COSIGN_PUBLIC_KEY//<br>/\\n          }")"
 
 ## Join ghcr integration yaml
 if [[ -n "${IMAGE+x}" && -n "${IMAGEPULLSECRET+x}" ]]; then
-  yq '. *+ load("tests/integration/var-img.yaml")' tests/integration/ghcr-values.yaml > ghcr-tmp
-  envsubst < ghcr-tmp > ghcr-values
+  yq '. *+ load("tests/integration/var-img.yaml")' tests/integration/ghcr-values.yaml >ghcr-tmp
+  envsubst <ghcr-tmp >ghcr-values
+  envsubst <tests/integration/ghcr-values.yaml >ghcr-validator
   rm ghcr-tmp
 else
-  echo "" > ghcr-values
+  echo "" >ghcr-values
 fi
 
 ### SINGLE TEST CASE ####################################
 single_test() { # ID TXT TYP REF NS MSG RES
   echo -n "[$1] $2"
-  i=0 # intialize iterator
-  while : ; do
-    i=$((i+1))
+  i=0                                                              # intialize iterator
+  export RAND=$(head -c 5 /dev/urandom | hexdump -ve '1/1 "%.2x"') # creating a random index to label the pods and avoid name collision for repeated runs
```

Comment:
Would it be a better solution to simply set CTR=0 at start of file and inside single_test CTR+=1? Then we cant have random collisions fail the test and it works for arbitrarily many tests

Reply:
that would actually only work as long we are worried about collisions within this single run of the script. However, if we re-run a different test later, we will definitely get collisions. Therefore, I'd prefer to stick to the random one

Reply:
👍